src/parser/RdfParser.cpp

```diff
@@ -466,6 +468,23 @@ TripleComponent TurtleParser<T>::literalAndDatatypeToTripleComponentImpl(
       asNormalizedStringViewUnsafe(normalizedLiteralContent));
   std::string_view type = asStringViewUnsafe(typeIri.getContent());
 
+  // Helper to handle literals that are invalid for the respective datatype
+  auto makeNormalLiteral = [parser, literal, normalizedLiteralContent,
+                            type](std::optional<std::exception> error =
+                                      std::nullopt) {
+    std::string errorMsg = "";
+    if (error.has_value()) {
+      errorMsg = error.value().what();
+    }
```

Comment:
```suggestion
    std::string_view errorMsg = error.value_or("");
    std::string_view sep = error.has_value() ? ": " : "";
```

Reply:
I have now changed the first line because this suggestion leads to a compiler error.

```
RdfParser.cpp:475:22: error: conversion from ‘std::exception’ to non-scalar type ‘std::string_view’ {aka ‘std::basic_string_view<char>’} requested
#15 258.6   475 |     std::string_view errorMsg = error.value_or("");
```